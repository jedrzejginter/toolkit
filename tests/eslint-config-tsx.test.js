const { ESLint } = require('eslint');
const getRules = require('./test-utils');

jest.mock('../eslint-env', () => {
  return {
    hasTypescript: true,
    hasReact: true,
    hasJest: false,
    hasNext: false,
    hasTestingLibrary: false,
  };
});

test('config for tsx file', async () => {
  expect(await getRules('index.tsx')).toMatchSnapshot();
});
