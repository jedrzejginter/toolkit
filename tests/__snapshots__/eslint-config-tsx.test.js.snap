// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config for tsx file 1`] = `
Array [
  "@typescript-eslint/brace-style: off",
  "@typescript-eslint/camelcase: off",
  "@typescript-eslint/comma-dangle: off",
  "@typescript-eslint/comma-spacing: off",
  "@typescript-eslint/dot-notation: error",
  "@typescript-eslint/func-call-spacing: off",
  "@typescript-eslint/indent: off",
  "@typescript-eslint/keyword-spacing: off",
  "@typescript-eslint/lines-between-class-members: error",
  "@typescript-eslint/member-delimiter-style: off",
  "@typescript-eslint/naming-convention: error",
  "@typescript-eslint/no-array-constructor: error",
  "@typescript-eslint/no-dupe-class-members: error",
  "@typescript-eslint/no-empty-function: error",
  "@typescript-eslint/no-extra-parens: off",
  "@typescript-eslint/no-extra-semi: off",
  "@typescript-eslint/no-implied-eval: error",
  "@typescript-eslint/no-loop-func: error",
  "@typescript-eslint/no-magic-numbers: off",
  "@typescript-eslint/no-redeclare: error",
  "@typescript-eslint/no-shadow: error",
  "@typescript-eslint/no-throw-literal: error",
  "@typescript-eslint/no-unused-expressions: error",
  "@typescript-eslint/no-unused-vars: error",
  "@typescript-eslint/no-use-before-define: error",
  "@typescript-eslint/no-useless-constructor: error",
  "@typescript-eslint/object-curly-spacing: off",
  "@typescript-eslint/quotes: off",
  "@typescript-eslint/semi: off",
  "@typescript-eslint/space-before-function-paren: off",
  "@typescript-eslint/space-infix-ops: off",
  "@typescript-eslint/type-annotation-spacing: off",
  "accessor-pairs: off",
  "array-bracket-newline: off",
  "array-bracket-spacing: off",
  "array-callback-return: error",
  "array-element-newline: off",
  "arrow-body-style: error",
  "arrow-parens: off",
  "arrow-spacing: off",
  "block-scoped-var: error",
  "block-spacing: off",
  "brace-style: off",
  "callback-return: off",
  "camelcase: off",
  "capitalized-comments: off",
  "class-methods-use-this: error",
  "comma-dangle: off",
  "comma-spacing: off",
  "comma-style: off",
  "complexity: off",
  "computed-property-spacing: off",
  "consistent-return: error",
  "consistent-this: off",
  "constructor-super: off",
  "curly: off",
  "default-case-last: off",
  "default-case: off",
  "default-param-last: off",
  "dot-location: off",
  "dot-notation: off",
  "eol-last: off",
  "eqeqeq: error",
  "for-direction: error",
  "func-call-spacing: off",
  "func-name-matching: off",
  "func-names: warn",
  "func-style: off",
  "function-call-argument-newline: off",
  "function-paren-newline: off",
  "generator-star-spacing: off",
  "generator-star: off",
  "getter-return: off",
  "global-require: error",
  "grouped-accessor-pairs: off",
  "guard-for-in: error",
  "handle-callback-err: off",
  "id-blacklist: off",
  "id-denylist: off",
  "id-length: off",
  "id-match: off",
  "implicit-arrow-linebreak: off",
  "import/default: off",
  "import/dynamic-import-chunkname: off",
  "import/export: error",
  "import/exports-last: off",
  "import/extensions: error",
  "import/first: error",
  "import/group-exports: off",
  "import/imports-first: off",
  "import/max-dependencies: off",
  "import/named: off",
  "import/namespace: off",
  "import/newline-after-import: error",
  "import/no-absolute-path: error",
  "import/no-amd: error",
  "import/no-anonymous-default-export: off",
  "import/no-commonjs: off",
  "import/no-cycle: error",
  "import/no-default-export: off",
  "import/no-deprecated: off",
  "import/no-duplicates: error",
  "import/no-dynamic-require: error",
  "import/no-extraneous-dependencies: error",
  "import/no-internal-modules: off",
  "import/no-mutable-exports: error",
  "import/no-named-as-default-member: error",
  "import/no-named-as-default: error",
  "import/no-named-default: error",
  "import/no-named-export: off",
  "import/no-namespace: off",
  "import/no-nodejs-modules: off",
  "import/no-relative-parent-imports: off",
  "import/no-restricted-paths: off",
  "import/no-self-import: error",
  "import/no-unassigned-import: off",
  "import/no-unresolved: off",
  "import/no-unused-modules: off",
  "import/no-useless-path-segments: error",
  "import/no-webpack-loader-syntax: error",
  "import/order: error",
  "import/prefer-default-export: off",
  "import/unambiguous: off",
  "indent-legacy: off",
  "indent: off",
  "init-declarations: off",
  "jsx-a11y/accessible-emoji: error",
  "jsx-a11y/alt-text: error",
  "jsx-a11y/anchor-has-content: error",
  "jsx-a11y/anchor-is-valid: off",
  "jsx-a11y/aria-activedescendant-has-tabindex: error",
  "jsx-a11y/aria-props: error",
  "jsx-a11y/aria-proptypes: error",
  "jsx-a11y/aria-role: error",
  "jsx-a11y/aria-unsupported-elements: error",
  "jsx-a11y/autocomplete-valid: off",
  "jsx-a11y/click-events-have-key-events: off",
  "jsx-a11y/control-has-associated-label: error",
  "jsx-a11y/heading-has-content: error",
  "jsx-a11y/html-has-lang: error",
  "jsx-a11y/iframe-has-title: error",
  "jsx-a11y/img-redundant-alt: error",
  "jsx-a11y/interactive-supports-focus: error",
  "jsx-a11y/label-has-associated-control: error",
  "jsx-a11y/label-has-for: off",
  "jsx-a11y/lang: error",
  "jsx-a11y/media-has-caption: error",
  "jsx-a11y/mouse-events-have-key-events: error",
  "jsx-a11y/no-access-key: error",
  "jsx-a11y/no-autofocus: error",
  "jsx-a11y/no-distracting-elements: error",
  "jsx-a11y/no-interactive-element-to-noninteractive-role: error",
  "jsx-a11y/no-noninteractive-element-interactions: off",
  "jsx-a11y/no-noninteractive-element-to-interactive-role: error",
  "jsx-a11y/no-noninteractive-tabindex: off",
  "jsx-a11y/no-onchange: off",
  "jsx-a11y/no-redundant-roles: error",
  "jsx-a11y/no-static-element-interactions: off",
  "jsx-a11y/role-has-required-aria-props: error",
  "jsx-a11y/role-supports-aria-props: error",
  "jsx-a11y/scope: error",
  "jsx-a11y/tabindex-no-positive: error",
  "jsx-quotes: off",
  "key-spacing: off",
  "keyword-spacing: off",
  "line-comment-position: off",
  "linebreak-style: off",
  "lines-around-comment: off",
  "lines-around-directive: error",
  "lines-between-class-members: off",
  "max-classes-per-file: error",
  "max-depth: off",
  "max-len: off",
  "max-lines-per-function: off",
  "max-lines: off",
  "max-nested-callbacks: off",
  "max-params: off",
  "max-statements-per-line: off",
  "max-statements: off",
  "multiline-comment-style: off",
  "multiline-ternary: off",
  "new-cap: error",
  "new-parens: off",
  "newline-after-var: off",
  "newline-before-return: off",
  "newline-per-chained-call: off",
  "no-alert: warn",
  "no-array-constructor: off",
  "no-arrow-condition: off",
  "no-async-promise-executor: error",
  "no-await-in-loop: error",
  "no-bitwise: error",
  "no-buffer-constructor: error",
  "no-caller: error",
  "no-case-declarations: error",
  "no-catch-shadow: off",
  "no-class-assign: error",
  "no-comma-dangle: off",
  "no-compare-neg-zero: error",
  "no-cond-assign: error",
  "no-confusing-arrow: off",
  "no-console: warn",
  "no-const-assign: off",
  "no-constant-condition: warn",
  "no-constructor-return: off",
  "no-continue: error",
  "no-control-regex: error",
  "no-debugger: error",
  "no-delete-var: error",
  "no-div-regex: off",
  "no-dupe-args: off",
  "no-dupe-class-members: off",
  "no-dupe-else-if: off",
  "no-dupe-keys: off",
  "no-duplicate-case: error",
  "no-duplicate-imports: off",
  "no-else-return: error",
  "no-empty-character-class: error",
  "no-empty-function: off",
  "no-empty-pattern: error",
  "no-empty: error",
  "no-eq-null: off",
  "no-eval: error",
  "no-ex-assign: error",
  "no-extend-native: error",
  "no-extra-bind: error",
  "no-extra-boolean-cast: error",
  "no-extra-label: error",
  "no-extra-parens: off",
  "no-extra-semi: off",
  "no-fallthrough: error",
  "no-floating-decimal: off",
  "no-func-assign: off",
  "no-global-assign: error",
  "no-implicit-coercion: off",
  "no-implicit-globals: off",
  "no-implied-eval: off",
  "no-import-assign: off",
  "no-inline-comments: off",
  "no-inner-declarations: error",
  "no-invalid-regexp: error",
  "no-invalid-this: off",
  "no-irregular-whitespace: error",
  "no-iterator: error",
  "no-label-var: error",
  "no-labels: error",
  "no-lone-blocks: error",
  "no-lonely-if: error",
  "no-loop-func: off",
  "no-loss-of-precision: off",
  "no-magic-numbers: off",
  "no-misleading-character-class: error",
  "no-mixed-operators: off",
  "no-mixed-requires: off",
  "no-mixed-spaces-and-tabs: off",
  "no-multi-assign: error",
  "no-multi-spaces: off",
  "no-multi-str: error",
  "no-multiple-empty-lines: off",
  "no-native-reassign: off",
  "no-negated-condition: off",
  "no-negated-in-lhs: off",
  "no-nested-ternary: off",
  "no-new-func: off",
  "no-new-object: error",
  "no-new-require: error",
  "no-new-symbol: off",
  "no-new-wrappers: error",
  "no-new: error",
  "no-obj-calls: off",
  "no-octal-escape: error",
  "no-octal: error",
  "no-param-reassign: off",
  "no-path-concat: error",
  "no-plusplus: error",
  "no-process-env: off",
  "no-process-exit: off",
  "no-promise-executor-return: off",
  "no-proto: error",
  "no-prototype-builtins: error",
  "no-redeclare: off",
  "no-regex-spaces: error",
  "no-reserved-keys: off",
  "no-restricted-exports: off",
  "no-restricted-globals: off",
  "no-restricted-imports: off",
  "no-restricted-modules: off",
  "no-restricted-properties: error",
  "no-restricted-syntax: off",
  "no-return-assign: error",
  "no-return-await: error",
  "no-script-url: error",
  "no-self-assign: error",
  "no-self-compare: error",
  "no-sequences: error",
  "no-setter-return: off",
  "no-shadow-restricted-names: error",
  "no-shadow: off",
  "no-space-before-semi: off",
  "no-spaced-func: off",
  "no-sparse-arrays: error",
  "no-sync: off",
  "no-tabs: off",
  "no-template-curly-in-string: error",
  "no-ternary: off",
  "no-this-before-super: off",
  "no-throw-literal: off",
  "no-trailing-spaces: off",
  "no-undef-init: error",
  "no-undef: off",
  "no-undefined: off",
  "no-underscore-dangle: off",
  "no-unexpected-multiline: off",
  "no-unmodified-loop-condition: off",
  "no-unneeded-ternary: error",
  "no-unreachable-loop: off",
  "no-unreachable: off",
  "no-unsafe-finally: error",
  "no-unsafe-negation: off",
  "no-unused-expressions: off",
  "no-unused-labels: error",
  "no-unused-vars: off",
  "no-use-before-define: off",
  "no-useless-backreference: off",
  "no-useless-call: off",
  "no-useless-catch: error",
  "no-useless-computed-key: error",
  "no-useless-concat: error",
  "no-useless-constructor: off",
  "no-useless-escape: error",
  "no-useless-rename: error",
  "no-useless-return: error",
  "no-var: error",
  "no-void: error",
  "no-warning-comments: off",
  "no-whitespace-before-property: off",
  "no-with: error",
  "no-wrap-func: off",
  "nonblock-statement-body-position: off",
  "object-curly-newline: off",
  "object-curly-spacing: off",
  "object-property-newline: off",
  "object-shorthand: error",
  "one-var-declaration-per-line: off",
  "one-var: error",
  "operator-assignment: error",
  "operator-linebreak: off",
  "padded-blocks: off",
  "padding-line-between-statements: off",
  "prefer-arrow-callback: error",
  "prefer-const: error",
  "prefer-destructuring: error",
  "prefer-exponentiation-operator: off",
  "prefer-named-capture-group: off",
  "prefer-numeric-literals: error",
  "prefer-object-spread: error",
  "prefer-promise-reject-errors: error",
  "prefer-reflect: off",
  "prefer-regex-literals: off",
  "prefer-rest-params: error",
  "prefer-spread: error",
  "prefer-template: error",
  "prettier/prettier: error",
  "quote-props: off",
  "quotes: off",
  "radix: error",
  "react-hooks/exhaustive-deps: error",
  "react-hooks/rules-of-hooks: error",
  "react/boolean-prop-naming: off",
  "react/button-has-type: error",
  "react/default-props-match-prop-types: error",
  "react/destructuring-assignment: off",
  "react/display-name: off",
  "react/forbid-component-props: off",
  "react/forbid-dom-props: off",
  "react/forbid-elements: off",
  "react/forbid-foreign-prop-types: warn",
  "react/forbid-prop-types: error",
  "react/function-component-definition: off",
  "react/jsx-boolean-value: error",
  "react/jsx-child-element-spacing: off",
  "react/jsx-closing-bracket-location: off",
  "react/jsx-closing-tag-location: off",
  "react/jsx-curly-brace-presence: error",
  "react/jsx-curly-newline: off",
  "react/jsx-curly-spacing: off",
  "react/jsx-equals-spacing: off",
  "react/jsx-filename-extension: error",
  "react/jsx-first-prop-new-line: off",
  "react/jsx-fragments: error",
  "react/jsx-handler-names: off",
  "react/jsx-indent-props: off",
  "react/jsx-indent: off",
  "react/jsx-key: off",
  "react/jsx-max-depth: off",
  "react/jsx-max-props-per-line: off",
  "react/jsx-newline: off",
  "react/jsx-no-bind: error",
  "react/jsx-no-comment-textnodes: error",
  "react/jsx-no-duplicate-props: error",
  "react/jsx-no-literals: off",
  "react/jsx-no-script-url: off",
  "react/jsx-no-target-blank: error",
  "react/jsx-no-undef: error",
  "react/jsx-no-useless-fragment: off",
  "react/jsx-one-expression-per-line: off",
  "react/jsx-pascal-case: error",
  "react/jsx-props-no-multi-spaces: off",
  "react/jsx-props-no-spreading: off",
  "react/jsx-sort-default-props: off",
  "react/jsx-sort-prop-types: off",
  "react/jsx-sort-props: off",
  "react/jsx-space-before-closing: off",
  "react/jsx-tag-spacing: off",
  "react/jsx-uses-react: error",
  "react/jsx-uses-vars: error",
  "react/jsx-wrap-multilines: off",
  "react/no-access-state-in-setstate: error",
  "react/no-adjacent-inline-elements: off",
  "react/no-array-index-key: off",
  "react/no-children-prop: error",
  "react/no-danger-with-children: error",
  "react/no-danger: off",
  "react/no-deprecated: error",
  "react/no-did-mount-set-state: off",
  "react/no-did-update-set-state: error",
  "react/no-direct-mutation-state: off",
  "react/no-find-dom-node: error",
  "react/no-is-mounted: error",
  "react/no-multi-comp: off",
  "react/no-redundant-should-component-update: error",
  "react/no-render-return-value: error",
  "react/no-set-state: off",
  "react/no-string-refs: error",
  "react/no-this-in-sfc: error",
  "react/no-typos: error",
  "react/no-unescaped-entities: off",
  "react/no-unknown-property: error",
  "react/no-unsafe: off",
  "react/no-unused-prop-types: error",
  "react/no-unused-state: error",
  "react/no-will-update-set-state: error",
  "react/prefer-es6-class: error",
  "react/prefer-read-only-props: off",
  "react/prefer-stateless-function: error",
  "react/prop-types: off",
  "react/react-in-jsx-scope: error",
  "react/require-default-props: off",
  "react/require-optimization: off",
  "react/require-render-return: error",
  "react/self-closing-comp: error",
  "react/sort-comp: error",
  "react/sort-prop-types: off",
  "react/state-in-constructor: off",
  "react/static-property-placement: off",
  "react/style-prop-object: error",
  "react/void-dom-elements-no-children: error",
  "require-atomic-updates: off",
  "require-await: off",
  "require-jsdoc: off",
  "require-unicode-regexp: off",
  "require-yield: error",
  "rest-spread-spacing: off",
  "semi-spacing: off",
  "semi-style: off",
  "semi: off",
  "sort-imports: off",
  "sort-keys: off",
  "sort-vars: off",
  "space-after-function-name: off",
  "space-after-keywords: off",
  "space-before-blocks: off",
  "space-before-function-paren: off",
  "space-before-function-parentheses: off",
  "space-before-keywords: off",
  "space-in-brackets: off",
  "space-in-parens: off",
  "space-infix-ops: off",
  "space-return-throw-case: off",
  "space-unary-ops: off",
  "space-unary-word-ops: off",
  "spaced-comment: error",
  "strict: error",
  "switch-colon-spacing: off",
  "symbol-description: error",
  "template-curly-spacing: off",
  "template-tag-spacing: off",
  "unicode-bom: off",
  "use-isnan: error",
  "valid-jsdoc: off",
  "valid-typeof: off",
  "vars-on-top: error",
  "wrap-iife: off",
  "wrap-regex: off",
  "yield-star-spacing: off",
  "yoda: error",
]
`;
